include/fmt/chrono.h

```diff
@@ -290,36 +291,39 @@ inline null<> localtime_s(...) { return null<>(); }
 inline null<> gmtime_r(...) { return null<>(); }
 inline null<> gmtime_s(...) { return null<>(); }
 
-template <typename Char>
-inline auto do_write(const std::tm& time, const std::locale& loc, char format,
-                     char modifier) -> std::basic_string<Char> {
-  auto&& os = std::basic_ostringstream<Char>();
-  os.imbue(loc);
-  using iterator = std::ostreambuf_iterator<Char>;
-  const auto& facet = std::use_facet<std::time_put<Char, iterator>>(loc);
-  auto end = facet.put(os, os, Char(' '), &time, format, modifier);
-  if (end.failed()) FMT_THROW(format_error("failed to format time"));
-  return std::move(os).str();
-}
-
-template <typename Char, typename OutputIt,
-          FMT_ENABLE_IF(!std::is_same<Char, char>::value)>
-auto write(OutputIt out, const std::tm& time, const std::locale& loc,
-           char format, char modifier = 0) -> OutputIt {
-  auto str = do_write<Char>(time, loc, format, modifier);
-  return std::copy(str.begin(), str.end(), out);
-}
-
 inline const std::locale& get_classic_locale() {
   static const auto& locale = std::locale::classic();
   return locale;
 }
 
-template <typename Char, typename OutputIt,
-          FMT_ENABLE_IF(std::is_same<Char, char>::value)>
-auto write(OutputIt out, const std::tm& time, const std::locale& loc,
-           char format, char modifier = 0) -> OutputIt {
-  auto str = do_write<char>(time, loc, format, modifier);
+template <typename CodeUnit> struct codecvt_result {
+  CodeUnit buf[32];
+  CodeUnit* end{};
+};
+
+template <typename CodeUnit>
+inline void write_codecvt(codecvt_result<CodeUnit>& out, string_view in_buf,
```

Comment:
Why not return `codecvt_result` instead of passing it as an output parameter?

Also inline is probably not needed here.

Reply:
``codecvt_result.end`` is depends on ``codecvt_result.buf``
Returning ``codecvt_result`` from a function may result in ``codecvt_result`` being copied and ``end`` being corrupted.
It is possible to replace the pointer ``end`` with the length ``len``, but this does not guarantee that there is no copying.

Reply:
Makes sense, let's leave as out parameter then.